website/api/controllers/webhooks/receive-from-zapier.js

```diff
@@ -0,0 +1,175 @@
+        return 'couldNotMatchLinkedinId';
+      });
+
+      // (FUTURE: make field for this and have it already in CRM so this step isn't necessary)
+      let linkedinCompanyId = matchedCompanyPageInfo.id;
+
+      // Check if we have enough space in our current active campaign.
+      // If so, then use it and update its inventory.  Otherwise, prepare to create
+      // a new campaign, and use that instead, updating our set of active campaigns
+      // in the db, including marking the new one as the latest and greatest.  Along the way,
+      // communicate with Campaign Manager to update or create the appropriate campaign.
+      let latestCampaign = await AdCampaign.findOne({ isLatest: true, persona: data.persona });
+      if (latestCampaign && latestCampaign.linkedinCompanyIds.length < 100) {
+        // Update ad campaign in Campaign Manager
+        // > For help w/ Linkedin API, see https://github.com/fleetdm/confidential/tree/main/ads
+        let filterCriteriaForLatestCampaign = latestCampaign.linkedinCompanyIds.map((id)=>{
+          return `urn:li:organization:${id}`;
+        });
+        await sails.helpers.http.sendHttpRequest.with({
+          method: 'POST',
+          url: `https://hooks.zapier.com/hooks/catch/3627242/2wdx23r?webhookSecret=${ sails.config.custom.zapierWebhookSecret}`,
+          body: {
+            campaignGroup: sails.config.custom.linkedinAbmCampaignGroupUrn,
+            name: latestCampaign.name,
+            linkedinCampaignUrn: latestCampaign.linkedinCampaignUrn,
+            targetingCriteria: filterCriteriaForLatestCampaign,
+          }
+        }).retry();
+
+        await AdCampaign.updateOne({ id: latestCampaign.id }).set({
+          linkedinCompanyIds: _.uniq(latestCampaign.linkedinCompanyIds.concat(linkedinCompanyId))
+        });
+      } else {
+
+        // First, mark the old campaign as no longer the latest.
+        // Note: Since we might not have done the first-time setup for
+        // this persona yet, it's possible there won't actually be a
+        // campaign record yet.  (In that case, we'll create it momentarily.)
+        if (latestCampaign) {
+          await AdCampaign.updateOne({ id: latestCampaign.id }).set({
+            isLatest: false,
+          });
+        }//ﬁ
+
+        // Create a placeholder linkedinCampaignUrn value to create the record with initially
+        // We'll use this value in a subsequent webhook run that will save update the record with the real linkedinCampaignUrn (once it has been created).
+        // Note: there is a possibility that a new campaign can't be created with only one linkedInCompanyID, (There is a minimum audience size of 300)
+        // In this case, we will treat this new campaign as the latest campaign in the website's database, and send updates for it as new company IDs are added.
+        // When the campaign actually exists in LinkedIn, Zapier will send another event to update the campaign urn in the website's database.
```

Comment:
FYI @mikermcneil 
The minimum audience size for new campaigns is 300, we will need to store incomplete campaigns in the websites database until the audience size is large enough to create a campaign. When that happens, a zap will send a request to update the record in the website's database record with the campaign's real URN.